src/utils/routes/PrivateRoute.tsx

```diff
@@ -1,12 +1,22 @@
-import { ComponentType } from 'react';
-import { Route, RouteComponentProps, RouteProps } from 'react-router-dom';
+import { Route, RouteProps } from 'react-router-dom';
 import { Forbidden } from '../../pages/errorpages/Forbidden';
 
 interface PrivateRouteProps extends RouteProps {
-  component: ComponentType<RouteComponentProps<any>> | ComponentType<any>;
   isAuthorized: boolean;
 }
 
-export const PrivateRoute = ({ component: ProtectedComponent, isAuthorized, ...rest }: PrivateRouteProps) => (
-  <Route {...rest} render={(props) => (isAuthorized ? <ProtectedComponent {...props} {...rest} /> : <Forbidden />)} />
-);
+export const PrivateRoute = ({ component, children, isAuthorized, ...rest }: PrivateRouteProps) => {
+  if (!isAuthorized) {
+    return (
+      <Route {...rest}>
+        <Forbidden />;
+      </Route>
+    );
+  } else if (component) {
+    return <Route {...rest} component={component} />;
+  } else if (children) {
+    return <Route {...rest}>{children}</Route>;
+  } else {
+    return null;
+  }
+};
```

Comment:
Tror vi kan forenkle dette litt mot det vi brukte som flyt først for å ha noe som virka. Kan det stemme at dette fungerer likt?
```suggestion
export const PrivateRoute = ({ isAuthorized, ...rest }: PrivateRouteProps) => {
  if (!isAuthorized) {
    return (
      <Route {...rest}>
        <Forbidden />;
      </Route>
    );
  }
  return <Route {...rest} />;
};
```

Reply:
Eller enda bedre:
```
export const PrivateRoute = ({ isAuthorized, ...rest }: PrivateRouteProps) =>
  isAuthorized ? <Route {...rest} /> : <Route {...rest} />;
```

Reply:
👌